### PR TITLE
DAOS-3411 control: record results of each system stop request

### PR DESCRIPTION
**Pull request by tanabarr**
SystemStop gRPC from mgmt tool will trigger Stop gRPCs to each system
member (resulting in retried host-local KillRank requests over dRPC).

KillRank will subsequently be replaced by InstanceStop (TBD) requests
to provide graceful shutdown capabilities (incl. inhibition of pool
rebuild).

This change adds SystemMemberResults to SystemStop response.
SystemMemberResults consist of id/errored/message for Stop requests
sent to each system member and enables reporting of individual or
aggregated instance stop request results to user.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/common/system_rpc.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import (
+	"errors"
+	"net"
+
+	mgmtpb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/daos-stack/daos/src/control/logging"
+)
+
+// MembersToPB converts internal member structs to protobuf equivalents.
+func MembersToPB(members SystemMembers) []*mgmtpb.SystemMember {
```
Comment (kjacque) [inline src/control/common/system_rpc.go]: Wondering if this conversion logic and SystemMember related stuff really belongs in common. Not something to change right now, but something to think about in the future. Feels sort of similar to security, where the domain is specific but used by both client and server.

Comment (tanabarr) [reply src/control/common/system_rpc.go]: do you think it needs its own package?

Comment (kjacque) [reply src/control/common/system_rpc.go]: That's what I was thinking. Or some package including only content related to these client/server communications. But I'm not sure if there's enough content to justify breaking it out, we'd need to look into it.

Comment (tanabarr) [reply src/control/common/system_rpc.go]: may well be warranted in the future

**Pull request merged**